### PR TITLE
Make `.get_data_as_epoch` timeout a configurable parameter

### PR DESCRIPTION
**Pull request by charlesincharge**
As someone creating a real-time application, I sometimes want to limit the amount of time the application is waiting on data. This change makes `wait_time` (now called `timeout`) a configurable parameter.

In the situation that data is not ready before the timeout, `.get_data_as_epoch` now returns `None` because `EpochsArray(data,...)` errors out when `data` is an empty array.
Note: it was possible for `data` to be empty even before the change, although a shorter `timeout` now makes this more likely.

### CODE REVIEW

**Comment by larsoner** [comment]
> In the situation that data is not ready before the timeout, .get_data_as_epoch now returns None because EpochsArray(data,...) errors out when data is an empty array.

Should we change MNE-Python to allow `EpochsArray` to be an empty array of shape `(0, n_channels, n_times)`? We do allow instances of empty `Epochs` so I don't see why we couldn't do it for `EpochsArray`...

**Comment by charlesincharge** [comment]
> Should we change MNE-Python to allow EpochsArray to be an empty array of shape (0, n_channels, n_times)? We do allow instances of empty Epochs so I don't see why we couldn't do it for EpochsArray...

Seems reasonable to me! The specific error raised here is because the array is empty with shape `(1, n_channels, n_times=0)`, but `EpochsArray` initialization also errors out if the passed in data is shape `(0, n_channels, n_times)` as you mention.

Separately, looks like the jobs are failing due to https://community.codecov.com/t/codecov-yanked-from-pypi-all-versions/4259/9

**Comment by larsoner** [comment]
The update I pushed should fix the codecov problem

Feel free to make a PR to allow 0 length EpochsArray. Maybe not for no time points, but it seems reasonable for zero epochs

File: mne_realtime/lsl_client.py [hunk 1]
```diff
@@ -48,34 +48,42 @@ class LSLClient(_BaseClient):
...
-        """Return last n_samples from current time.
+    def get_data_as_epoch(self, n_samples=1024, picks=None, timeout=None):
+        """Return n_samples from LSL in FIFO order.
 
         Parameters
         ----------
         n_samples : int
             Number of samples to fetch.
         %(picks_all)s
+        timeout : Optional[float]
```
Comment (larsoner) [inline mne_realtime/lsl_client.py:59]: Please use NumpyDoc format
```suggestion
        timeout : float | None
```

**Review by larsoner — COMMENTED** [review]
Other than one more numpydoc formatting thing, LGTM +1 for merge!

**Comment by larsoner** [comment]
Marking for merge when green @charlesincharge , thanks in advance!

**Pull request merged**